src/cairoUtilFuncGen/memory/memoryRead.ts

```diff
@@ -1,57 +1,46 @@
-import endent from 'endent';
 import {
   Expression,
   TypeName,
   FunctionCall,
   DataLocation,
-  FunctionStateMutability,
   generalizeType,
   TypeNode,
 } from 'solc-typed-ast';
 import { CairoFunctionDefinition, typeNameFromTypeNode } from '../../export';
 import {
   CairoFelt,
   CairoType,
-  CairoUint,
   MemoryLocation,
   TypeConversionContext,
 } from '../../utils/cairoTypeSystem';
 import { cloneASTNode } from '../../utils/cloning';
-import { createCairoGeneratedFunction, createCallToFunction } from '../../utils/functionGeneration';
-import {
-  ACCESSOR,
-  ACCESSOR_TRAIT,
-  WARPLIB_MEMORY,
-  WM_READ_FELT,
-  WM_READ_ID,
-} from '../../utils/importPaths';
-import { createNumberLiteral, createNumberTypeName } from '../../utils/nodeTemplates';
+import { createCallToFunction } from '../../utils/functionGeneration';
+import { WM_READ, WM_GET_ID, WM_RETRIEVE } from '../../utils/importPaths';
+import { createFeltLiteral, createFeltTypeName } from '../../utils/nodeTemplates';
 import { isDynamicArray, safeGetNodeType } from '../../utils/nodeTypeProcessing';
-import { GeneratedFunctionInfo, locationIfComplexType, StringIndexedFuncGen } from '../base';
+import { locationIfComplexType, StringIndexedFuncGen } from '../base';
 
 /*
   Produces functions that when given a start location in warp_memory, deserialise all necessary
   felts to produce a full value. For example, a function to read a Uint256 reads the given location
   and the next one, and combines them into a Uint256 struct
 */
-// TODO: Replace the whole class with a call to warp_memory.retrieve?
```

Comment:
The class doesn't generate any cairo code, only returns an instance of imported function now, right? Thus in future it shouldn't be a generated function. Am I right or I miss something?

Reply:
Yes you are right. We could remove it in the near future, once serde is proven to be sufficient for whatever type we would like to read